pcsx2/CDVD/ChdFileReader.cpp

```diff
@@ -16,71 +16,98 @@
 #include "PrecompiledHeader.h"
 #include "ChdFileReader.h"
 
-#include "CDVD/CompressedFileReaderUtils.h"
+#include "common/FileSystem.h"
+#include "common/StringUtil.h"
 
-#include <wx/dir.h>
+ChdFileReader::~ChdFileReader()
+{
+	Close();
+
+	for (std::FILE* fp : m_files)
+		std::fclose(fp);
+}
 
-bool ChdFileReader::CanHandle(const wxString& fileName)
+bool ChdFileReader::CanHandle(const std::string& fileName, const std::string& displayName)
 {
-	if (!wxFileName::FileExists(fileName) || !fileName.Lower().EndsWith(L".chd"))
-	{
+	if (!StringUtil::EndsWith(displayName, ".chd"))
 		return false;
-	}
+
 	return true;
 }
 
-bool ChdFileReader::Open2(const wxString& fileName)
+static chd_error chd_open_wrapper(const char* filename, std::FILE** fp, int mode, chd_file* parent, chd_file** chd)
+{
+	*fp = FileSystem::OpenCFile(filename, "rb");
+	if (!*fp)
+		return CHDERR_FILE_NOT_FOUND;
+
+	const chd_error err = chd_open_file(*fp, mode, parent, chd);
+	if (err == CHDERR_NONE)
+		return err;
+
+	std::fclose(*fp);
+	*fp = nullptr;
+	return err;
+}
+
+bool ChdFileReader::Open2(std::string fileName)
 {
 	Close2();
 
-	m_filename = fileName;
+	m_filename = std::move(fileName);
 
-	chd_file* child = NULL;
-	chd_file* parent = NULL;
+	chd_file* child = nullptr;
+	chd_file* parent = nullptr;
+	std::FILE* fp = nullptr;
 	chd_header header;
 	chd_header parent_header;
 
-	wxString chds[8];
-	chds[0] = fileName;
+	std::string chds[8];
+	chds[0] = m_filename;
 	int chd_depth = 0;
 	chd_error error;
 
-	// TODO: Unicode correctness on Windows
-	while (CHDERR_REQUIRES_PARENT == (error = chd_open(chds[chd_depth].c_str(), CHD_OPEN_READ, NULL, &child)))
+	std::string dirname;
+	FileSystem::FindResultsArray results;
+
+	while (CHDERR_REQUIRES_PARENT == (error = chd_open_wrapper(chds[chd_depth].c_str(), &fp, CHD_OPEN_READ, NULL, &child)))
 	{
 		if (chd_depth >= static_cast<int>(std::size(chds) - 1))
 		{
-			Console.Error(L"CDVD: chd_open hit recursion limit searching for parents");
+			Console.Error("CDVD: chd_open hit recursion limit searching for parents");
 			return false;
 		}
+
+		// TODO: This is still broken on Windows. Needs to be fixed in libchdr.
```

Comment:
I would suggest leaving a link to libchdr issue in the comment so it's easier to track. https://github.com/rtissera/libchdr/issues/55